<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Supplement to: "HaDeX: Analysis and Visualisation of Hydrogen/Deuterium Exchange Mass Spectrometry Data" • HaDeX</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Supplement to: " hadex: analysis and visualisation of hydrogen exchange mass spectrometry data>
<meta property="og:description" content="HaDeX">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">HaDeX</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.2.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/HaDeX.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/datafiles.html">Data processing</a>
    </li>
    <li>
      <a href="../articles/mass_comparison.html">Mass calculation</a>
    </li>
    <li>
      <a href="../articles/transformation.html">Data aggregation</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="HaDeX_files/htmlwidgets-1.5.2/htmlwidgets.js"></script><script src="HaDeX_files/jquery-1.12.4/jquery.min.js"></script><link href="HaDeX_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="HaDeX_files/datatables-binding-0.16/datatables.js"></script><link href="HaDeX_files/dt-core-bootstrap-1.10.20/css/dataTables.bootstrap.min.css" rel="stylesheet">
<link href="HaDeX_files/dt-core-bootstrap-1.10.20/css/dataTables.bootstrap.extra.css" rel="stylesheet">
<script src="HaDeX_files/dt-core-bootstrap-1.10.20/js/jquery.dataTables.min.js"></script><script src="HaDeX_files/dt-core-bootstrap-1.10.20/js/dataTables.bootstrap.min.js"></script><link href="HaDeX_files/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet">
<script src="HaDeX_files/crosstalk-1.1.0.1/js/crosstalk.min.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Supplement to: “HaDeX: Analysis and Visualisation of Hydrogen/Deuterium Exchange Mass Spectrometry Data”</h1>
                        <h4 class="author">Weronika Puchała, Michał Burdukiewicz, Michał Kistowski, Katarzyna A. Dąbrowska, Aleksandra E. Badaczewska-Dawid, Dominik Cysewski, Michał Dadlez</h4>
            
            <h4 class="date">20.06.2019</h4>
      
      
      <div class="hidden name"><code>HaDeX.Rmd</code></div>

    </div>

    
    
<div id="about-hadex" class="section level1">
<h1 class="hasAnchor">
<a href="#about-hadex" class="anchor"></a>About HaDeX</h1>
<p>HaDeX is a novel tool for processing, analysis, and visualization of HDX-MS experiments. HaDeX covers the final parts of the analytic process, including a comparison of experiments, quality control and generation of publication-quality figures. To make the HaDeX <strong>R</strong> package available to the less <strong>R</strong>-fluent users, we enhanced it with a comprehensive Graphical User Interface available as a HaDeX GUI. The reproducibility of the whole procedure is ensured with advanced reporting functions.</p>
<p>The GUI is available online: <a href="http://mslab-ibb.pl/shiny/HaDeX/" class="uri">http://mslab-ibb.pl/shiny/HaDeX/</a> or can be installed locally on Windows systems: <a href="https://sourceforge.net/projects/hadex/files/HaDeX_setup.exe/download" class="uri">https://sourceforge.net/projects/hadex/files/HaDeX_setup.exe/download</a>. Alternatively, R-fluent users can access the GUI through the <a href="https://hadexversum.github.io/HaDeX/reference/HaDeX_gui.html">HaDeX_gui()</a> function.</p>
<p>This document covers the main functionalities of both the <strong>R</strong> package and the GUI.</p>
<div id="comparison-of-the-existing-hdx-ms-software" class="section level2">
<h2 class="hasAnchor">
<a href="#comparison-of-the-existing-hdx-ms-software" class="anchor"></a>Comparison of the existing HDX-MS software</h2>
<p>To show the novelty of HaDeX, we compare its functionalities with other relatively new software for analysis of HDX-MS data: MEMHDX <span class="citation">(Hourdel et al. 2016)</span> and Deuteros <span class="citation">(Lau et al. 2019)</span>.</p>
<div id="htmlwidget-e957fe803be51078df82" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-e957fe803be51078df82">{"x":{"style":"bootstrap","filter":"none","data":[["Accessibility","Accessibility","Accessibility","Analysis","Analysis","Analysis","Analysis","Analysis","Analysis","Analysis","Analysis","Visualization","Visualization","Visualization","Visualization","Visualization","Visualization","Visualization","Visualization","Visualization","Reporting","Reporting","Reporting","Reporting","Reporting"],["Web server","Programmatic access","Desktop software","Multi-state analysis","ISO-based uncertainty","Coverage and peptide overlap","Kinetics analysis","Quality control","N- and C-terminal length corrections","Wald's test","Consolidation","Global visualization of deuterium uptake","Woods plot","Butterfly plot","Logit plot","Zooming of the Woods plot","Customizable label names and colors","Peptide kinetics chart","3D structure visualization","Downloadable charts","Deuterium uptake download","Downloadable results of intermediate computations","Report generation","PyMol export","HDX Data Summary"],["Yes","No","No","Yes","No","Yes","No","No","Yes","No","No","Yes","Yes","No","No","No","Yes","Yes","Yes","Yes","No","No","No","No","No"],["Yes","No","No","No","No","No","Yes","No","No","Yes","Yes","Yes","No","Yes","Yes","No","No","Yes","Yes","Yes","Yes","No","No","No","No"],["No","No","Yes","No","No","No","No","No","Yes","No","No","Yes","Yes","No","Yes","Yes","No","No","No","Yes","Yes","No","No","Yes","No"],["Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","No","No","Yes","Yes","No","No","Yes","Yes","Yes","No","Yes","Yes","Yes","Yes","No","Yes"]],"container":"<table class=\"table table-striped table-hover row-border order-column display\">\n  <thead>\n    <tr>\n      <th>Type<\/th>\n      <th>Name<\/th>\n      <th>MSTools<\/th>\n      <th>MEMHDX<\/th>\n      <th>Deuteros<\/th>\n      <th>HaDeX<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"t","ordering":false,"paging":false,"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data) {\nvar value=data[2]; $(this.api().cell(row, 2).node()).css({'background-color':value == \"Yes\" ? \"#00BFFF\" : value == \"No\" ? \"#FF8C91\" : null});\nvar value=data[3]; $(this.api().cell(row, 3).node()).css({'background-color':value == \"Yes\" ? \"#00BFFF\" : value == \"No\" ? \"#FF8C91\" : null});\nvar value=data[4]; $(this.api().cell(row, 4).node()).css({'background-color':value == \"Yes\" ? \"#00BFFF\" : value == \"No\" ? \"#FF8C91\" : null});\nvar value=data[5]; $(this.api().cell(row, 5).node()).css({'background-color':value == \"Yes\" ? \"#00BFFF\" : value == \"No\" ? \"#FF8C91\" : null});\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script><p>We have not considered HDX Workbench <span class="citation">(Pascal et al. 2012)</span> as it deals with the preliminary steps of the analysis. This comparison also does not cover a versatile structural visualisation tool for HDX-MS results, HDX-Viewer <span class="citation">(Bouyssié et al. 2019)</span> as its scope is different from the tools mentioned above.</p>
<p><strong>Web server:</strong> a software is available as a web server.</p>
<p><strong>Programmatic access:</strong> analytic functionalities are documented and available from a command line.</p>
<p><strong>Desktop software:</strong> a software can be installed locally.</p>
<p><strong>Multi-state analysis:</strong> a software supports comparisons of more than two states.</p>
<p><strong>ISO-based uncertainty:</strong> analytic functions produce ISO-compatible uncertainty intervals.</p>
<p><strong>Coverage and peptide overlap:</strong> overview of experimental sequence coverage is available in a user-friendly way.</p>
<p><strong>Quality control:</strong> additional information about course of the experiment.</p>
<p><strong>N- and C-terminal length corrections:</strong> manual correction of sequence length.</p>
<p><strong>Global visualization of deuterium uptake:</strong> deuterium uptake for different states is shown together for comparison.</p>
<p><strong>Woods plot:</strong> deuteration difference between chosen states shown in the format of Woods plot.</p>
<p><strong>Zooming of the Woods plot:</strong> Woods plot can be zoomed in.</p>
<p><strong>Customizable label names and colors:</strong> Labels and colors on the plot can be changed by the user.</p>
<p><strong>Peptide kinetics chart:</strong> Kinetics plot (deuteration change in time) are available for each peptide.</p>
<p><strong>3D structure visualization:</strong> structure of the protein is visualized in 3D.</p>
<p><strong>Downloadable charts:</strong> charts are downloadable, preferably in a vector format (.eps, .svg or .pdf).</p>
<p><strong>Deuterium uptake download:</strong> data shown on Woods plot is downloadable (e.q. as CSV file).</p>
<p><strong>Downloadable results of intermediate computations:</strong> results of intermediate computations (e.q. pure deuteration data) are downloadable.</p>
<p><strong>Report generation:</strong> generates a report (e.q. in Html format) with results of the analysis with parametrization.</p>
<p><strong>PyMol export:</strong> exports data to the <a href="https://pymol.org/2/">PyMol</a> format.</p>
<p><strong>HDX Data Summary:</strong> summary of the experimental data <span class="citation">(Masson et al. 2019)</span>.</p>
</div>
</div>
<div id="hadex-functionalities" class="section level1">
<h1 class="hasAnchor">
<a href="#hadex-functionalities" class="anchor"></a>HaDeX functionalities</h1>
<div id="data-import" class="section level2">
<h2 class="hasAnchor">
<a href="#data-import" class="anchor"></a>Data import</h2>
<p>The HaDeX web server works only on data in the DynamX datafile format (Waters Corp.). The data from other sources may be also adjusted to the format accepted by HaDeX provided it has following columns:</p>
<div id="htmlwidget-bd9155998b8ede50e002" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-bd9155998b8ede50e002">{"x":{"style":"bootstrap","filter":"none","data":[["Protein","Start","End","Sequence","Modification","Fragment","MaxUptake","MHP","State","Exposure","File","z","RT","Inten","Center"],["Character","Integer","Integer","Character","Logic","Logic","Numeric","Numeric","Character","Numeric","Character","Integer","Numeric","Numeric","Numeric"]],"container":"<table class=\"table table-striped table-hover row-border order-column display\">\n  <thead>\n    <tr>\n      <th>Column.Name<\/th>\n      <th>Column.Type<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"t","ordering":false,"paging":false,"autoWidth":true,"order":[],"orderClasses":false}},"evals":[],"jsHooks":[]}</script><p>Although data can be imported into R using other tools, we strongly advise to rely on the <a href="https://hadexversum.github.io/HaDeX/reference/read_hdx.html">read_hdx()</a> function:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_hdx.html">read_hdx</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span>package <span class="op">=</span> <span class="st">"HaDeX"</span>, 
                            <span class="st">"HaDeX/data/KD_190304_Nucb2_EDTA_CaCl2_test02_clusterdata.csv"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>Currently, <a href="https://hadexversum.github.io/HaDeX/reference/read_hdx.html">read_hdx()</a> supports .csv, .tsv and .xls files fulfilling the data structure described above. Files with data from multiple proteins are supported. The user need to indicate which protein is of interest for calculations.</p>
</div>
<div id="computation-of-deuteration-levels" class="section level2">
<h2 class="hasAnchor">
<a href="#computation-of-deuteration-levels" class="anchor"></a>Computation of deuteration levels</h2>
<p>The computation of the level of deuteration involves several pre-processing steps, all of which are described in this section. These steps are performed automatically in the GUI or by the <a href="https://hadexversum.github.io/HaDeX/reference/prepare_dataset.html">prepare_dataset()</a> function in the console.</p>
<div id="measured-data-into-overall-peptide-mass" class="section level3">
<h3 class="hasAnchor">
<a href="#measured-data-into-overall-peptide-mass" class="anchor"></a>Measured data into overall peptide mass</h3>
<p>The results of HDX-MS measurements as given in the DynamX data files are represented as the measured mass of peptides plus proton mass to charge ratio (<span class="math inline">\(Center\)</span>). For later use, this value has to be transformed into an overall mass of a peptide measured after specific time point from a protein in a specific state, as shown in equation 1:</p>
<p><span class="math display">\[pepMass = z \times (Center-protonMass)\tag{1}\]</span> where:</p>
<ul>
<li><p><span class="math inline">\(pepMass\)</span> - expected mass of the peptide after incubation (Da),</p></li>
<li><p><span class="math inline">\(protonMass\)</span> - the mass of the proton (Da),</p></li>
<li><p><span class="math inline">\(z\)</span> - charge of the peptide,</p></li>
<li><p><span class="math inline">\(Center\)</span> - experimentally measured peptide mass plus proton mass to charge ratio <span class="math inline">\(\left(\frac{m}{z}\right)\)</span>.</p></li>
</ul>
<p>HDX-MS experiments are often repeated (by the rule of thumb at least three times). Thus, we aggregate the results of replicates as a weighted mean mass into a single result per peptide using equation 2:</p>
<p><span class="math display">\[aggMass = \sum_{k = 1}^{N}\frac{Inten_k}{N}{\times}pepMass_k\tag{2}\]</span> where:</p>
<ul>
<li><p><span class="math inline">\(aggMass\)</span> - weighted mean mass of the peptide (Da),</p></li>
<li><p><span class="math inline">\(k\)</span> - replicate index,</p></li>
<li><p><span class="math inline">\(Inten\)</span> - intensity,</p></li>
<li><p><span class="math inline">\(N\)</span> - number of replicates.</p></li>
</ul>
<p>This data manipulation results from an original data structure. Each repetition of the measurement gives the data for given peptide in a given time per possible value of <span class="math inline">\(z\)</span> as shown in the example below. We need to use the value of <span class="math inline">\(pepMass\)</span> as shown in equation 1 but still keep the information about original measurement - that’s why we use the weighted mass.</p>
<pre><code>##                 Sequence                            File z       RT  Inten
## 1 KQFEHLNHQNPDTFEPKDLDML KD_190119_gg_Nucb2_CaCl2_10s_01 3 5.396196  95419
## 2 KQFEHLNHQNPDTFEPKDLDML KD_190119_gg_Nucb2_CaCl2_10s_01 4 5.392023 194073
##     Center
## 1 901.7158
## 2 676.4348</code></pre>
<p>The uncertainty of a measurement is variability associated with the precision of measuring instrumentation. We present here a novel derivation of uncertainty formulas for HDX-MS data according to the ISO guidelines <span class="citation">(Joint Committee for Guides in Metrology 2008)</span>. Input files always encompass results of more than one measurement. We assume uncorrelatedness of replicates as they come from different samples. Therefore, we average measurements of replicates for each time point and for all protein states. Thus, we compute peptide mass uncertainty <span class="math inline">\(u\)</span> as uncertainty for aggregate estimate using the formula for standard deviation of the mean:</p>
<p><span class="math display">\[u(x) = \sqrt{\frac{ \sum_{i=1}^n \left( x_{i} - \overline{x} \right)^2}{n(n-1)}}\tag{3}\]</span> where:</p>
<ul>
<li><p><span class="math inline">\(x_{i}\)</span> - measurement,</p></li>
<li><p><span class="math inline">\(\overline{x}\)</span> - mean value,</p></li>
<li><p><span class="math inline">\(n\)</span> - number of measurements.</p></li>
</ul>
<p>After obtaining the mass of the peptide, we can compute the deuteration level depending on the chosen maximum deuteration level. The maximum deuteration can also be computed in two different ways: either as <em>theoretical</em> (where the maximum deuteration depends on the theoretical deuteration levels) and <em>experimental</em> (where the maximum deuteration is assumed to be equal to the deuteration measured at the last time point).</p>
</div>
<div id="experimental-deuteration-level" class="section level3">
<h3 class="hasAnchor">
<a href="#experimental-deuteration-level" class="anchor"></a>Experimental deuteration level</h3>
<p>The experimental deuteration level is computed as the deuteration level of the peptide from a protein in a specific state and after incubation time <span class="math inline">\(t\)</span> compared to the deuteration level measured at the start of the incubation (<span class="math inline">\(t_0\)</span>). It yields a value for the chosen state and chosen time <span class="math inline">\(t\)</span>.</p>
<p><span class="math display">\[D = D_{t} - D_{t_0}\tag{4}\]</span></p>
<p>where:</p>
<ul>
<li><p><span class="math inline">\(D\)</span> - deuteration level (Da),</p></li>
<li><p><span class="math inline">\(D_{t_0}\)</span> - experimentally measured deuteration at the beginning of the incubation (0 or close to 0),</p></li>
</ul>
<p>The equation 4 produces only absolute deuteration levels. The computations of relative deuteration levels follows a similar logic and is normalized by the difference of deuteration between the start (<span class="math inline">\(t_0\)</span>) and the end of the experiment (<span class="math inline">\(t_out\)</span>) as shown in the equation 4a:</p>
<p><span class="math display">\[D = \frac{D_{t} - D_{t_0}}{D_{t_{out}} - D_{t_0}}\tag{4a}\]</span></p>
<p>All functions in the HaDeX package contain the logical parameter <span class="math inline">\(relative\)</span> to determine if they should return absolute or relative deuteration levels.</p>
<div id="uncertainty-calculations" class="section level4">
<h4 class="hasAnchor">
<a href="#uncertainty-calculations" class="anchor"></a>Uncertainty calculations</h4>
<p>We describe the methodology of the uncertainty calculations for relative deuteration levels. The uncertainty for absolute deuteration levels is computed similarly, but without scaling.</p>
<p>To calculate uncertainty related to functions of more than one variables (e. g., equation 4) the Law of propagation of uncertainty is defined by equation 5:</p>
<p><span class="math display">\[u_{c}(y) = \sqrt{\sum_{k} \left[ \frac{\partial y}{\partial x_{k}} u(x_{k}) \right]^2}\tag{5}\]</span></p>
<p>As the variable of interest is <span class="math inline">\(D\)</span>, we apply the general formula to the deuteration level <span class="math inline">\(D\)</span> (equation 6):</p>
<p><span class="math display">\[u_{c}(D) = \sqrt{\sum_{k} \left[ \frac{\partial D}{\partial D_{k}} u(D_{k}) \right]^2 }\tag{6}\]</span></p>
<p>where:</p>
<ul>
<li><p><span class="math inline">\(k \in \{0, t, out\}\)</span>,</p></li>
<li><p><span class="math inline">\(D_{k}\)</span> - deuteration in <span class="math inline">\(k\)</span> time (Da),</p></li>
<li><p><span class="math inline">\(u(D_{k})\)</span> - an uncertainty associated with <span class="math inline">\(D_{k}\)</span> as standard deviation of the mean value,</p></li>
</ul>
<p>Then, expanding the equation 6:</p>
<p><span class="math display">\[u_{c}(D) = \sqrt{ \left[ 
\frac{1}{D_{t_{out}}-D_{t_0}} u(D_{t}) \right]^2 + 
\left[ \frac{D_{t} - D_{t_{out}}}{(D_{t_{out}}-D_{t_0})^2} u(D_{t_0}) \right]^2 + 
\left[ \frac{D_{t_0} - D_{t}}{(D_{t_{out}}-D_{t_0})^2} u (D_{t_{out}}) \right]^2}\tag{7}\]</span> As expected, the uncertainty associated with <span class="math inline">\(D_{t}\)</span> has the biggest impact on <span class="math inline">\(u_{c}(D)\)</span>.</p>
</div>
</div>
<div id="theoretical-deuteration-level" class="section level3">
<h3 class="hasAnchor">
<a href="#theoretical-deuteration-level" class="anchor"></a>Theoretical deuteration level</h3>
<p>As opposed to the experimental deuteration levels, theoretical deuteration level only partially depends on the experimental data. Here, the maximum deuteration level is based on a hypothetical peptide where all hydrogens were replaced by deuterons, as it is shown in equation 8:</p>
<p><span class="math display">\[D = \frac{D_{t}-MHP}{MaxUptake \times protonMass}\tag{8}\]</span></p>
<p>where:</p>
<ul>
<li><p><span class="math inline">\(D_{t}\)</span> - deuteration measured in a chosen time point (Da),</p></li>
<li><p><span class="math inline">\(MHP\)</span> - theoretical mass of the peptide (constant) (Da),</p></li>
<li><p><span class="math inline">\(MaxUptake\)</span> - the maximum proton uptake for the peptide (theoretical constant) (Da),</p></li>
<li><p><span class="math inline">\(protonMass\)</span> - mass of a proton (constant) (Da).</p></li>
</ul>
<p>The absolute deuteration level is calculated as in equation 8 but without scaling (equation 8a):</p>
<p><span class="math display">\[D = D_{t} - MHP\tag{8a}\]</span></p>
<div id="uncertainty-calculations-1" class="section level4">
<h4 class="hasAnchor">
<a href="#uncertainty-calculations-1" class="anchor"></a>Uncertainty calculations</h4>
<p>For functions of one variable uncertainty reduces to:</p>
<p><span class="math display">\[u(y) = \left| \frac{dy}{dx} u(x) \right|.\tag{9}\]</span></p>
<p>Substituting <span class="math inline">\(D\)</span> from equation 8, we have</p>
<p><span class="math display">\[u(D) = \left|\frac{1}{MaxUptake \times protonMass} u(D_{t}) \right|\tag{10}\]</span></p>
<p>For the absolute values, <span class="math inline">\(u(D)\)</span> is identical with <span class="math inline">\(u(D_{t})\)</span>, based on equations 8a and 9.</p>
</div>
</div>
</div>
<div id="difference-of-deuteration-levels-between-two-states" class="section level2">
<h2 class="hasAnchor">
<a href="#difference-of-deuteration-levels-between-two-states" class="anchor"></a>Difference of deuteration levels between two states</h2>
<p>The differences of deuteration levels between two states are associated with a different level of protection of hydrogens. Therefore, we are especially interested in the differential analysis of the deuteration levels. Thus, the deuteration level in one state <span class="math inline">\((D_{2})\)</span> is subtracted from deuteration level in the other state <span class="math inline">\((D_{1})\)</span>:</p>
<p><span class="math display">\[diff = D_{1} - D_{2}\tag{11}\]</span></p>
<p>and the uncertainty is a function of two variables (based on equation 11 and 5):</p>
<p><span class="math display">\[u_{c}(diff) = \sqrt{u(D_{1})^2 + u(D_{2})^2}\tag{12}\]</span></p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">calc_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prepare_dataset.html">prepare_dataset</a></span><span class="op">(</span><span class="va">dat</span>,
                            in_state_first <span class="op">=</span> <span class="st">"gg_Nucb2_EDTA_0.001"</span>,
                            chosen_state_first <span class="op">=</span> <span class="st">"gg_Nucb2_EDTA_25"</span>,
                            out_state_first <span class="op">=</span> <span class="st">"gg_Nucb2_EDTA_1440"</span>,
                            in_state_second <span class="op">=</span> <span class="st">"gg_Nucb2_CaCl2_0.001"</span>,
                            chosen_state_second <span class="op">=</span> <span class="st">"gg_Nucb2_CaCl2_25"</span>,
                            out_state_second <span class="op">=</span> <span class="st">"gg_Nucb2_CaCl2_1440"</span><span class="op">)</span> </code></pre></div>
</div>
<div id="visual-data-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#visual-data-analysis" class="anchor"></a>Visual data analysis</h2>
<div id="comparison-of-states" class="section level3">
<h3 class="hasAnchor">
<a href="#comparison-of-states" class="anchor"></a>Comparison of states</h3>
<p>Comparison plots show the deuteration level of all peptides in selected states in a given time. The x-axis represents positions of amino acids in the sequence. The y-axis shows the deuteration level, expressed either as a <em>relative</em> or <em>absolute</em> deuteration. The chart below shows peptide deuteration after 25 minutes of the incubation along with the confidence intervals.</p>
<p>– theoretical:</p>
<ul>
<li>relative values:</li>
</ul>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/comparison_plot.html">comparison_plot</a></span><span class="op">(</span>calc_dat <span class="op">=</span> <span class="va">calc_dat</span>,
                theoretical <span class="op">=</span> <span class="cn">TRUE</span>,
                relative <span class="op">=</span> <span class="cn">TRUE</span>,
                state_first <span class="op">=</span> <span class="st">"Nucb2 Factor 1"</span>,
                state_second <span class="op">=</span> <span class="st">"Nucb2 Factor 2"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Theoretical fraction exchanged in state comparison in 25 min time"</span><span class="op">)</span></code></pre></div>
<p><img src="HaDeX_files/figure-html/unnamed-chunk-7-1.png" width="672"></p>
<ul>
<li>absolute values:</li>
</ul>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/comparison_plot.html">comparison_plot</a></span><span class="op">(</span>calc_dat <span class="op">=</span> <span class="va">calc_dat</span>,
                theoretical <span class="op">=</span> <span class="cn">TRUE</span>,
                relative <span class="op">=</span> <span class="cn">FALSE</span>,
                state_first <span class="op">=</span> <span class="st">"Nucb2 Factor 1"</span>,
                state_second <span class="op">=</span> <span class="st">"Nucb2 Factor 2"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Theoretical fraction exchanged in state comparison in 25 min time"</span><span class="op">)</span></code></pre></div>
<p><img src="HaDeX_files/figure-html/unnamed-chunk-8-1.png" width="672"></p>
<p>– experimental:</p>
<ul>
<li>relative values:</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/comparison_plot.html">comparison_plot</a></span><span class="op">(</span>calc_dat <span class="op">=</span> <span class="va">calc_dat</span>,
                theoretical <span class="op">=</span> <span class="cn">FALSE</span>,
                relative <span class="op">=</span> <span class="cn">TRUE</span>,
                state_first <span class="op">=</span> <span class="st">"Nucb2 Factor 1"</span>,
                state_second <span class="op">=</span> <span class="st">"Nucb2 Factor 2"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Fraction exchanged in state comparison in 25 min time"</span><span class="op">)</span></code></pre></div>
<p><img src="HaDeX_files/figure-html/unnamed-chunk-9-1.png" width="672"></p>
<ul>
<li>absolute values:</li>
</ul>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/comparison_plot.html">comparison_plot</a></span><span class="op">(</span>calc_dat <span class="op">=</span> <span class="va">calc_dat</span>,
                theoretical <span class="op">=</span> <span class="cn">FALSE</span>,
                relative <span class="op">=</span> <span class="cn">FALSE</span>, 
                state_first <span class="op">=</span> <span class="st">"Nucb2 Factor 1"</span>,
                state_second <span class="op">=</span> <span class="st">"Nucb2 Factor 2"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Fraction exchanged in state comparison in 25 min time"</span><span class="op">)</span></code></pre></div>
<p><img src="HaDeX_files/figure-html/unnamed-chunk-10-1.png" width="672"></p>
</div>
</div>
<div id="woods-plot" class="section level2">
<h2 class="hasAnchor">
<a href="#woods-plot" class="anchor"></a>Woods plot</h2>
<p>Woods plots show the difference between the deuteration of all peptides in two different states in a specific time point as described by equation 11. Similarly to the comparison plot, HaDeX provides both experimental and theoretical deuteration levels using either relative or absolute values:</p>
<p>– theoretical:</p>
<ul>
<li>relative values:</li>
</ul>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/woods_plot.html">woods_plot</a></span><span class="op">(</span>calc_dat <span class="op">=</span> <span class="va">calc_dat</span>,
           theoretical <span class="op">=</span> <span class="cn">TRUE</span>,
           relative <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Theoretical fraction exchanged between states in 25 min time"</span><span class="op">)</span></code></pre></div>
<p><img src="HaDeX_files/figure-html/unnamed-chunk-11-1.png" width="672"></p>
<ul>
<li>absolute values:</li>
</ul>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/woods_plot.html">woods_plot</a></span><span class="op">(</span>calc_dat <span class="op">=</span> <span class="va">calc_dat</span>,
           theoretical <span class="op">=</span> <span class="cn">TRUE</span>,
           relative <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Theoretical fraction exchanged between states in 25 min time"</span><span class="op">)</span></code></pre></div>
<p><img src="HaDeX_files/figure-html/unnamed-chunk-12-1.png" width="672"></p>
<p>– experimental:</p>
<ul>
<li>relative values:</li>
</ul>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/woods_plot.html">woods_plot</a></span><span class="op">(</span>calc_dat <span class="op">=</span> <span class="va">calc_dat</span>,
           theoretical <span class="op">=</span> <span class="cn">FALSE</span>, 
           relative <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Theoretical fraction exchanged between states in 25 min time"</span><span class="op">)</span></code></pre></div>
<p><img src="HaDeX_files/figure-html/unnamed-chunk-13-1.png" width="672"></p>
<ul>
<li>absolute values:</li>
</ul>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/woods_plot.html">woods_plot</a></span><span class="op">(</span>calc_dat <span class="op">=</span> <span class="va">calc_dat</span>,
           theoretical <span class="op">=</span> <span class="cn">FALSE</span>, 
           relative <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Theoretical fraction exchanged between states in 25 min time"</span><span class="op">)</span></code></pre></div>
<p><img src="HaDeX_files/figure-html/unnamed-chunk-14-1.png" width="672"></p>
<div id="confidence-limit-in-woods-plot" class="section level3">
<h3 class="hasAnchor">
<a href="#confidence-limit-in-woods-plot" class="anchor"></a>Confidence limit in Woods plot</h3>
<p>The function <a href="https://hadexversum.github.io/HaDeX/reference/calculate_confidence_limit_values.html">calculate_confidence_limit_values()</a> calculates confidence limit values as it is described elsewhere <span class="citation">(Houde, Berkowitz, and Engen 2011)</span>.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/calculate_confidence_limit_values.html">calculate_confidence_limit_values</a></span><span class="op">(</span>calc_dat <span class="op">=</span> <span class="va">calc_dat</span>,
                                  confidence_limit <span class="op">=</span> <span class="fl">0.99</span>,
                                  theoretical <span class="op">=</span> <span class="cn">FALSE</span>, 
                                  relative <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>  </code></pre></div>
<pre><code>## [1] -0.01619004  0.01619004</code></pre>
<p>The function <a href="https://hadexversum.github.io/HaDeX/reference/add_stat_dependency.html">add_stat_dependency()</a> returns data extended by column describing relation of a given peptide with confidence limit.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/add_stat_dependency.html">add_stat_dependency</a></span><span class="op">(</span><span class="va">calc_dat</span>, 
                   confidence_limit <span class="op">=</span> <span class="fl">0.98</span>, 
                   theoretical <span class="op">=</span> <span class="cn">FALSE</span>, 
                   relative <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 108 x 29
##    Sequence Start   End Med_Sequence frac_exch_state~ err_frac_exch_s~
##    &lt;chr&gt;    &lt;int&gt; &lt;int&gt;        &lt;dbl&gt;            &lt;dbl&gt;            &lt;dbl&gt;
##  1 VPIDID      17    22         19.5          NaN            NaN      
##  2 KTKVKGE~    23    44         33.5          NaN            NaN      
##  3 YYDEY       45    49         47              0.975          0.00984
##  4 YYDEYL      45    50         47.5            0.679          0.00549
##  5 YLRQVID     49    55         52              0.453          0.00309
##  6 YLRQVIDV    49    56         52.5            0.428          0.00341
##  7 YLRQVID~    49    57         53              0.417          0.00271
##  8 LRQVID      50    55         52.5            0.520          0.00660
##  9 LRQVIDV     50    56         53              0.450          0.00248
## 10 LRQVIDVL    50    57         53.5            0.414          0.00346
## # ... with 98 more rows, and 23 more variables: frac_exch_state_2 &lt;dbl&gt;,
## #   err_frac_exch_state_2 &lt;dbl&gt;, diff_frac_exch &lt;dbl&gt;, err_frac_exch &lt;dbl&gt;,
## #   abs_frac_exch_state_1 &lt;dbl&gt;, err_abs_frac_exch_state_1 &lt;dbl&gt;,
## #   abs_frac_exch_state_2 &lt;dbl&gt;, err_abs_frac_exch_state_2 &lt;dbl&gt;,
## #   abs_diff_frac_exch &lt;dbl&gt;, err_abs_diff_frac_exch &lt;dbl&gt;,
## #   avg_theo_in_time_1 &lt;dbl&gt;, err_avg_theo_in_time_1 &lt;dbl&gt;,
## #   avg_theo_in_time_2 &lt;dbl&gt;, err_avg_theo_in_time_2 &lt;dbl&gt;,
## #   diff_theo_frac_exch &lt;dbl&gt;, err_diff_theo_frac_exch &lt;dbl&gt;,
## #   abs_avg_theo_in_time_1 &lt;dbl&gt;, err_abs_avg_theo_in_time_1 &lt;dbl&gt;,
## #   abs_avg_theo_in_time_2 &lt;dbl&gt;, err_abs_avg_theo_in_time_2 &lt;dbl&gt;,
## #   abs_diff_theo_frac_exch &lt;dbl&gt;, err_abs_diff_theo_frac_exch &lt;dbl&gt;,
## #   valid_at_0.98 &lt;lgl&gt;</code></pre>
</div>
</div>
<div id="kinetic-plots" class="section level2">
<h2 class="hasAnchor">
<a href="#kinetic-plots" class="anchor"></a>Kinetic plots</h2>
<p>By the term <em>kinetics</em> we understand deuteration change in time. To calculate deuteration values per peptide for different time points is used function <a href="https://hadexversum.github.io/HaDeX/reference/calculate_kinetics.html">calculate_kinetics()</a>. This function uses the <a href="https://hadexversum.github.io/HaDeX/reference/calculate_state_deuteration.html">calculate_state_deuteration()</a> function.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="va">kin_YYDEYL_gg_Nucb2_CaCl2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculate_kinetics.html">calculate_kinetics</a></span><span class="op">(</span>dat <span class="op">=</span> <span class="va">dat</span>, 
                                                protein <span class="op">=</span> <span class="st">"db_Nucb2"</span>, 
                                                sequence <span class="op">=</span> <span class="st">"YYDEYL"</span>,
                                                state <span class="op">=</span> <span class="st">"gg_Nucb2_CaCl2"</span>, 
                                                start <span class="op">=</span> <span class="fl">45</span>, 
                                                end <span class="op">=</span> <span class="fl">50</span>, 
                                                time_in <span class="op">=</span> <span class="fl">0.001</span>, 
                                                time_out <span class="op">=</span> <span class="fl">1440</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 5 x 15
##   Protein Sequence Start   End State time_chosen frac_exch_state
##   &lt;chr&gt;   &lt;chr&gt;    &lt;int&gt; &lt;int&gt; &lt;chr&gt;       &lt;dbl&gt;           &lt;dbl&gt;
## 1 db_Nuc~ YYDEYL      45    50 gg_N~       0.167            20.9
## 2 db_Nuc~ YYDEYL      45    50 gg_N~       1                37.7
## 3 db_Nuc~ YYDEYL      45    50 gg_N~      10                74.1
## 4 db_Nuc~ YYDEYL      45    50 gg_N~      25                86.2
## 5 db_Nuc~ YYDEYL      45    50 gg_N~      60                93.9
## # ... with 8 more variables: err_frac_exch_state &lt;dbl&gt;,
## #   abs_frac_exch_state &lt;dbl&gt;, err_abs_frac_exch_state &lt;dbl&gt;,
## #   avg_theo_in_time &lt;dbl&gt;, err_avg_theo_in_time &lt;dbl&gt;,
## #   abs_avg_theo_in_time &lt;dbl&gt;, err_abs_avg_theo_in_time &lt;dbl&gt;,
## #   Med_Sequence &lt;dbl&gt;</code></pre>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="va">kin_YYDEYL_gg_Nucb2_EDTA</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculate_kinetics.html">calculate_kinetics</a></span><span class="op">(</span>dat <span class="op">=</span> <span class="va">dat</span>, 
                                              protein <span class="op">=</span> <span class="st">"db_Nucb2"</span>, 
                                              sequence <span class="op">=</span> <span class="st">"YYDEYL"</span>,
                                              state <span class="op">=</span> <span class="st">"gg_Nucb2_EDTA"</span>, 
                                              start <span class="op">=</span> <span class="fl">45</span>, 
                                              end <span class="op">=</span> <span class="fl">50</span>, 
                                              time_in <span class="op">=</span> <span class="fl">0.001</span>, 
                                              time_out <span class="op">=</span> <span class="fl">1440</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 5 x 15
##   Protein Sequence Start   End State time_chosen frac_exch_state
##   &lt;chr&gt;   &lt;chr&gt;    &lt;int&gt; &lt;int&gt; &lt;chr&gt;       &lt;dbl&gt;           &lt;dbl&gt;
## 1 db_Nuc~ YYDEYL      45    50 gg_N~       0.167            16.4
## 2 db_Nuc~ YYDEYL      45    50 gg_N~       1                34.0
## 3 db_Nuc~ YYDEYL      45    50 gg_N~      10                60.7
## 4 db_Nuc~ YYDEYL      45    50 gg_N~      25                67.9
## 5 db_Nuc~ YYDEYL      45    50 gg_N~      60                79.0
## # ... with 8 more variables: err_frac_exch_state &lt;dbl&gt;,
## #   abs_frac_exch_state &lt;dbl&gt;, err_abs_frac_exch_state &lt;dbl&gt;,
## #   avg_theo_in_time &lt;dbl&gt;, err_avg_theo_in_time &lt;dbl&gt;,
## #   abs_avg_theo_in_time &lt;dbl&gt;, err_abs_avg_theo_in_time &lt;dbl&gt;,
## #   Med_Sequence &lt;dbl&gt;</code></pre>
<p>Calculated data can be shown next to each other on the plot for comparison. To visualize kinetic data we recommend <a href="https://hadexversum.github.io/HaDeX/reference/plot_kinetics.html">plot_kinetics()</a> function:</p>
<p>– theoretical:</p>
<ul>
<li>relative values:</li>
</ul>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html">bind_rows</a></span><span class="op">(</span><span class="va">kin_YYDEYL_gg_Nucb2_CaCl2</span>, <span class="va">kin_YYDEYL_gg_Nucb2_EDTA</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="../reference/plot_kinetics.html">plot_kinetics</a></span><span class="op">(</span>theoretical <span class="op">=</span> <span class="cn">TRUE</span>, 
                relative <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="HaDeX_files/figure-html/unnamed-chunk-19-1.png" width="672"></p>
<ul>
<li>absolute values:</li>
</ul>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html">bind_rows</a></span><span class="op">(</span><span class="va">kin_YYDEYL_gg_Nucb2_CaCl2</span>, <span class="va">kin_YYDEYL_gg_Nucb2_EDTA</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="../reference/plot_kinetics.html">plot_kinetics</a></span><span class="op">(</span>theoretical <span class="op">=</span> <span class="cn">TRUE</span>, 
                relative <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p><img src="HaDeX_files/figure-html/unnamed-chunk-20-1.png" width="672"></p>
<p>– experimental:</p>
<ul>
<li>relative values:</li>
</ul>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html">bind_rows</a></span><span class="op">(</span><span class="va">kin_YYDEYL_gg_Nucb2_CaCl2</span>, <span class="va">kin_YYDEYL_gg_Nucb2_EDTA</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="../reference/plot_kinetics.html">plot_kinetics</a></span><span class="op">(</span>theoretical <span class="op">=</span> <span class="cn">FALSE</span>, 
                relative <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="HaDeX_files/figure-html/unnamed-chunk-21-1.png" width="672"></p>
<ul>
<li>absolute values:</li>
</ul>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html">bind_rows</a></span><span class="op">(</span><span class="va">kin_YYDEYL_gg_Nucb2_CaCl2</span>, <span class="va">kin_YYDEYL_gg_Nucb2_EDTA</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="../reference/plot_kinetics.html">plot_kinetics</a></span><span class="op">(</span>theoretical <span class="op">=</span> <span class="cn">FALSE</span>, 
                relative <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p><img src="HaDeX_files/figure-html/unnamed-chunk-22-1.png" width="672"></p>
</div>
<div id="additional-tools" class="section level2">
<h2 class="hasAnchor">
<a href="#additional-tools" class="anchor"></a>Additional tools</h2>
<p>HaDeX provides additional tools for assessment of experiments.</p>
<div id="peptide-coverage" class="section level3">
<h3 class="hasAnchor">
<a href="#peptide-coverage" class="anchor"></a>Peptide coverage</h3>
<p>The sequence of the protein(s) is reconstructed from the peptides from the input file. Thus, amino acids not covered by peptides are marked as X according to the IUPAC convention. The sequence is reconstructed using the <a href="https://hadexversum.github.io/HaDeX/reference/reconstruct_sequence.html">reconstruct_sequence()</a> function.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/reconstruct_sequence.html">reconstruct_sequence</a></span><span class="op">(</span><span class="va">dat</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "xxxxxxxxxxxxxxxxVPIDIDKTKVKGEGHVEGEKIENPDTGLYYDEYLRQVIDVLETDKHFREKLQTADIEEIKSGKLSRELDLVSHHVRTRLDELKRQEVARLRMLIKAKMDSVQDTGIDHQALLKQFEHLNHQNPDTFEPKDLDMLIKAATSDLENYDKTRHEEFKKYEMxxxxxxxxxxxxLDEEKRQREESKFGEMxxxxxxxxxxxxxxxxxxxKEVWEEADGLDPNEFDPKTFFKLHDVNNDRFLDEQELEAxFTKELEKVYDPKNEEDDMVEMEEERLxxxxHVMNEVDINKDRLVTLEEFLRATEKKEFLEPDSWETLDQQQLFTEDELKEFESHISQQEDELRKKAEELQKQKEELQRQHDQLQAQEQELQQVVKQMEQKKLQQANPPAGPAGELK"</code></pre>
<p>Additionally, the coverage of peptides can be presented on a chart using the <a href="https://hadexversum.github.io/HaDeX/reference/plot_coverage.html">plot_coverage()</a> and <a href="https://hadexversum.github.io/HaDeX/reference/plot_position_frequency.html">plot_position_frequency()</a> functions.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_coverage.html">plot_coverage</a></span><span class="op">(</span><span class="va">dat</span>, chosen_state <span class="op">=</span> <span class="st">"gg_Nucb2_CaCl2"</span><span class="op">)</span></code></pre></div>
<p><img src="HaDeX_files/figure-html/unnamed-chunk-24-1.png" width="672"></p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_position_frequency.html">plot_position_frequency</a></span><span class="op">(</span><span class="va">dat</span>, chosen_state <span class="op">=</span> <span class="st">"gg_Nucb2_CaCl2"</span><span class="op">)</span></code></pre></div>
<p><img src="HaDeX_files/figure-html/unnamed-chunk-24-2.png" width="672"></p>
<p>The user can choose which state (or states) should be included in these plots. If this parameter is not provided, the first possible state is chosen. If a given peptide is available in more than one state, it is shown only once.</p>
</div>
<div id="quality-control" class="section level3">
<h3 class="hasAnchor">
<a href="#quality-control" class="anchor"></a>Quality control</h3>
<p>The function <a href="https://hadexversum.github.io/HaDeX/reference/quality_control.html">quality_control()</a> plots the change in the uncertainty of deuteration levels as a function of incubation time. The uncertainty is averaged over all peptides available at a given time point in a selected state. This chart has a double function: firstly, it allows checking if the measurement uncertainty is decreasing over time (which is the expected behavior) and the second one helps to plan the appropriate incubation length for the tested protein (whether we obtain the desired data reliability values).</p>
<p>In HDX-MS experiments, the average uncertainty of measurements decreases over time as the sample is more and more deuterated (even accounting for the back-propagation). Therefore, if the level of uncertainty is stable, the exchange is still ongoing and it would be advisable to prolong the reaction.</p>
<p>Moreover, the user can detect a time point after which the decrease of the deuteration uncertainty becomes too marginal to prolong the measurements. This function is most useful in case of multiple measurements of the same or very similar proteins because it helps to optimize the duration of the incubation. The result of this function can be easily visualized.</p>
<p>Example:</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/quality_control.html">quality_control</a></span><span class="op">(</span>dat <span class="op">=</span> <span class="va">dat</span>,
                          state_first <span class="op">=</span> <span class="st">"gg_Nucb2_EDTA"</span>,
                          state_second <span class="op">=</span> <span class="st">"gg_Nucb2_CaCl2"</span>, 
                          chosen_time <span class="op">=</span> <span class="fl">1</span>, 
                          in_time <span class="op">=</span> <span class="fl">0.001</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">result</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html">geom_line</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">out_time</span>, y <span class="op">=</span> <span class="va">avg_err_state_first</span>, color <span class="op">=</span> <span class="st">"Average error (first state)"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html">geom_line</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">out_time</span>, y <span class="op">=</span> <span class="va">avg_err_state_second</span>, color <span class="op">=</span> <span class="st">"Average error (second state)"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_x_log10</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"log(time) [min]"</span>, y <span class="op">=</span> <span class="st">"Average uncertainty"</span>, title <span class="op">=</span> <span class="st">"Uncertainty change"</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span>base_size <span class="op">=</span> <span class="fl">11</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"bottom"</span>,
        legend.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="HaDeX_files/figure-html/unnamed-chunk-26-1.png" width="672"></p>
<p>The chart below shows how we interpret the three most common results of quality control.</p>
<p><img src="HaDeX_files/figure-html/unnamed-chunk-27-1.png" width="672"></p>
</div>
</div>
</div>
<div id="hadex-graphical-user-interface" class="section level1">
<h1 class="hasAnchor">
<a href="#hadex-graphical-user-interface" class="anchor"></a>HaDeX Graphical User Interface</h1>
<p>The HaDeX Shiny app is launched by the <a href="https://hadexversum.github.io/HaDeX/reference/HaDeX_gui.html">HaDeX_gui()</a> function or available at MS Lab website: <a href="http://mslab-ibb.pl/shiny/HaDeX/" class="uri">http://mslab-ibb.pl/shiny/HaDeX/</a>. In the Input Data tab the user can adjust parameters to be propagated for the whole analysis.</p>
</div>
<div id="examples" class="section level1">
<h1 class="hasAnchor">
<a href="#examples" class="anchor"></a>Examples</h1>
<div id="example-1-cd160-hvem" class="section level2">
<h2 class="hasAnchor">
<a href="#example-1-cd160-hvem" class="anchor"></a>Example 1: CD160-HVEM</h2>
<p>The interaction between HVEM and the <a href="https://www.uniprot.org/uniprot/O95971">CD160</a> receptor was measured with HDX-MS.</p>
<p>Firstly, we read the input data, exactly as provided by the DynamX 3.0 (Waters Corp.)</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">HaDeX</span><span class="op">)</span>

<span class="co"># file import</span>
<span class="va">dat_1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_hdx.html">read_hdx</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span>package <span class="op">=</span> <span class="st">"HaDeX"</span>, 
                              <span class="st">"HaDeX/data/KD_180110_CD160_HVEM.csv"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>Then, we reconstruct the protein sequence from the peptides measured during the experiment. We observe the region from amino acid 107 till amino acid 124 is not covered by any peptide.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/reconstruct_sequence.html">reconstruct_sequence</a></span><span class="op">(</span><span class="va">dat_1</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "INITSSASQEGTRLNLICTVWHKKEEAEGFVVFLCKDRSGDCSPETSLKQLRLKRDPGIDGVGEISSQLMFTISQVTPLHSGTYQCCARSQKSGIRLQGHFFSILFxxxxxxxxxxxxxxxxxxFSHNEGTL"</code></pre>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_position_frequency.html">plot_position_frequency</a></span><span class="op">(</span><span class="va">dat_1</span>, chosen_state <span class="op">=</span> <span class="st">"CD160"</span><span class="op">)</span></code></pre></div>
<p><img src="HaDeX_files/figure-html/unnamed-chunk-29-1.png" width="672"></p>
<p>The theoretical plot allows finding regions, which exchange quickly (N-terminus, regions between 30-70 amino acid) and regions, which exchange slowly (peptides 15-24, 95-110 amino acid). We can also see the differences in the exchange between two states, indicating regions which changed upon binding with other protein.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># calculate data</span>
<span class="va">calc_dat_1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prepare_dataset.html">prepare_dataset</a></span><span class="op">(</span>dat <span class="op">=</span> <span class="va">dat_1</span>,
                              in_state_first <span class="op">=</span> <span class="st">"CD160_0.001"</span>,
                              chosen_state_first <span class="op">=</span> <span class="st">"CD160_1"</span>,
                              out_state_first <span class="op">=</span> <span class="st">"CD160_1440"</span>,
                              in_state_second <span class="op">=</span> <span class="st">"CD160_HVEM_0.001"</span>,
                              chosen_state_second <span class="op">=</span> <span class="st">"CD160_HVEM_1"</span>,
                              out_state_second <span class="op">=</span> <span class="st">"CD160_HVEM_1440"</span><span class="op">)</span>                             

<span class="co"># theoretical comparison plot - relative values</span>
<span class="fu"><a href="../reference/comparison_plot.html">comparison_plot</a></span><span class="op">(</span>calc_dat <span class="op">=</span> <span class="va">calc_dat_1</span>,
                theoretical <span class="op">=</span> <span class="cn">TRUE</span>,
                relative <span class="op">=</span> <span class="cn">TRUE</span>, 
                state_first <span class="op">=</span> <span class="st">"CD160"</span>,
                state_second <span class="op">=</span> <span class="st">"CD160_HVEM"</span><span class="op">)</span></code></pre></div>
<p><img src="HaDeX_files/figure-html/unnamed-chunk-30-1.png" width="672"></p>
<p>On the experimental plot, there are visible regions, which exchange quickly (N terminal part, regions between 30-70 amino acid) and regions, which exchange slowly (peptides 15-24, 30-35, 95-110 amino acid).</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># experimental comparison plot - relative values</span>
<span class="fu"><a href="../reference/comparison_plot.html">comparison_plot</a></span><span class="op">(</span>calc_dat <span class="op">=</span> <span class="va">calc_dat_1</span>,
                theoretical <span class="op">=</span> <span class="cn">FALSE</span>,
                relative <span class="op">=</span> <span class="cn">TRUE</span>,
                state_first <span class="op">=</span> <span class="st">"CD160"</span>,
                state_second <span class="op">=</span> <span class="st">"CD160_HVEM"</span><span class="op">)</span></code></pre></div>
<p><img src="HaDeX_files/figure-html/unnamed-chunk-31-1.png" width="672"></p>
<p>Plots below are equivalent to plots above but in absolute values - for users that prefer those.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># theoretical comparison plot - absolute values</span>
<span class="fu"><a href="../reference/comparison_plot.html">comparison_plot</a></span><span class="op">(</span>calc_dat <span class="op">=</span> <span class="va">calc_dat_1</span>,
                theoretical <span class="op">=</span> <span class="cn">TRUE</span>,
                relative <span class="op">=</span> <span class="cn">FALSE</span>, 
                state_first <span class="op">=</span> <span class="st">"CD160"</span>,
                state_second <span class="op">=</span> <span class="st">"CD160_HVEM"</span><span class="op">)</span></code></pre></div>
<p><img src="HaDeX_files/figure-html/unnamed-chunk-32-1.png" width="672"></p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># experimental comparison plot - absolute values</span>
<span class="fu"><a href="../reference/comparison_plot.html">comparison_plot</a></span><span class="op">(</span>calc_dat <span class="op">=</span> <span class="va">calc_dat_1</span>,
                theoretical <span class="op">=</span> <span class="cn">FALSE</span>,
                relative <span class="op">=</span> <span class="cn">FALSE</span>,
                state_first <span class="op">=</span> <span class="st">"CD160"</span>,
                state_second <span class="op">=</span> <span class="st">"CD160_HVEM"</span><span class="op">)</span></code></pre></div>
<p><img src="HaDeX_files/figure-html/unnamed-chunk-32-2.png" width="672"></p>
<p>The plot below shows peptides for which levels of deuteration were significantly lower upon binding with other protein (red) and peptides for which levels of exchange were significantly higher upon binding with other protein (blue). The plot also shows peptides in which no significant changes in deuteration between states are visible (grey). The biggest changes on the theoretical plot can be observed in 3 peptides (15-24, 30-35, 75-90).</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># theoretical Woods plot - relative values</span>
<span class="fu"><a href="../reference/woods_plot.html">woods_plot</a></span><span class="op">(</span>calc_dat <span class="op">=</span> <span class="va">calc_dat_1</span>,
           theoretical <span class="op">=</span> <span class="cn">TRUE</span>, 
           relative <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_cartesian.html">coord_cartesian</a></span><span class="op">(</span>ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">.2</span>, <span class="fl">.2</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="HaDeX_files/figure-html/unnamed-chunk-33-1.png" width="672"></p>
<p>The biggest changes on the experimental plot can be observed in 3 peptides (15-24, 110-115).</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># experimental Woods plot - relative values</span>
<span class="fu"><a href="../reference/woods_plot.html">woods_plot</a></span><span class="op">(</span>calc_dat <span class="op">=</span> <span class="va">calc_dat_1</span>,
           theoretical <span class="op">=</span> <span class="cn">FALSE</span>,
           relative <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_cartesian.html">coord_cartesian</a></span><span class="op">(</span>ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">.2</span>, <span class="fl">.2</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="HaDeX_files/figure-html/unnamed-chunk-34-1.png" width="672"></p>
<p>Plot below shows results in absolute values.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># theoretical Woods plot - absolute values</span>
<span class="fu"><a href="../reference/woods_plot.html">woods_plot</a></span><span class="op">(</span>calc_dat <span class="op">=</span> <span class="va">calc_dat_1</span>,
           theoretical <span class="op">=</span> <span class="cn">TRUE</span>, 
           relative <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Theoretical fraction exchanged between states in 1 min time"</span><span class="op">)</span> </code></pre></div>
<p><img src="HaDeX_files/figure-html/unnamed-chunk-35-1.png" width="672"></p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># experimental Woods plot - absolute values</span>
<span class="fu"><a href="../reference/woods_plot.html">woods_plot</a></span><span class="op">(</span>calc_dat <span class="op">=</span> <span class="va">calc_dat_1</span>,
           theoretical <span class="op">=</span> <span class="cn">FALSE</span>,
           relative <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Fraction exchanged between states in 1 min time"</span><span class="op">)</span> </code></pre></div>
<p><img src="HaDeX_files/figure-html/unnamed-chunk-35-2.png" width="672"></p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># quality control - relative values </span>
<span class="op">(</span><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/quality_control.html">quality_control</a></span><span class="op">(</span>dat <span class="op">=</span> <span class="va">dat_1</span>,
                state_first <span class="op">=</span> <span class="st">"CD160"</span>,
                state_second <span class="op">=</span> <span class="st">"CD160_HVEM"</span>, 
                chosen_time <span class="op">=</span> <span class="fl">1</span>, 
                in_time <span class="op">=</span> <span class="fl">0.001</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>##   out_time avg_err_state_first sd_err_state_first avg_err_state_second
## 1        5         0.002925674        0.002207625          0.003811229
## 2       25         0.002086275        0.001636480          0.003375989
## 3      120         0.002124354        0.001284390          0.002759393
## 4     1440         0.001841383        0.001032972          0.002532581
##   sd_err_state_second    avg_diff     sd_diff
## 1         0.004561764 0.004920756 0.004952401
## 2         0.003690424 0.004053767 0.003949308
## 3         0.001976257 0.003470818 0.002088545
## 4         0.001120593 0.003200211 0.001369362</code></pre>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># example quality control visualisation</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">result</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html">geom_line</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">out_time</span>, y <span class="op">=</span> <span class="va">avg_err_state_first</span>, color <span class="op">=</span> <span class="st">"Average error (first state)"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html">geom_line</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">out_time</span>, y <span class="op">=</span> <span class="va">avg_err_state_second</span>, color <span class="op">=</span> <span class="st">"Average error (second state)"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_x_log10</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">ylim</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0.05</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"log(time) [min]"</span>, y <span class="op">=</span> <span class="st">"Average uncertainty"</span>, title <span class="op">=</span> <span class="st">"Uncertainty change in out time"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>,
        legend.position <span class="op">=</span> <span class="st">"bottom"</span><span class="op">)</span></code></pre></div>
<p><img src="HaDeX_files/figure-html/unnamed-chunk-35-3.png" width="672"></p>
</div>
<div id="example-workflow-2" class="section level2">
<h2 class="hasAnchor">
<a href="#example-workflow-2" class="anchor"></a>Example workflow 2</h2>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">HaDeX</span><span class="op">)</span>

<span class="co"># file import</span>
<span class="va">dat_2</span> <span class="op">&lt;-</span>  <span class="fu"><a href="../reference/read_hdx.html">read_hdx</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span>package <span class="op">=</span> <span class="st">"HaDeX"</span>, 
                               <span class="st">"HaDeX/data/KD_190304_Nucb2_EDTA_CaCl2_test02_clusterdata.csv"</span><span class="op">)</span><span class="op">)</span>

<span class="co"># protein sequence reconstruction</span>
<span class="fu"><a href="../reference/reconstruct_sequence.html">reconstruct_sequence</a></span><span class="op">(</span><span class="va">dat_2</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "xxxxxxxxxxxxxxxxVPIDIDKTKVKGEGHVEGEKIENPDTGLYYDEYLRQVIDVLETDKHFREKLQTADIEEIKSGKLSRELDLVSHHVRTRLDELKRQEVARLRMLIKAKMDSVQDTGIDHQALLKQFEHLNHQNPDTFEPKDLDMLIKAATSDLENYDKTRHEEFKKYEMxxxxxxxxxxxxLDEEKRQREESKFGEMxxxxxxxxxxxxxxxxxxxKEVWEEADGLDPNEFDPKTFFKLHDVNNDRFLDEQELEAxFTKELEKVYDPKNEEDDMVEMEEERLxxxxHVMNEVDINKDRLVTLEEFLRATEKKEFLEPDSWETLDQQQLFTEDELKEFESHISQQEDELRKKAEELQKQKEELQRQHDQLQAQEQELQQVVKQMEQKKLQQANPPAGPAGELK"</code></pre>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># calculate data</span>
<span class="va">calc_dat_2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prepare_dataset.html">prepare_dataset</a></span><span class="op">(</span>dat <span class="op">=</span> <span class="va">dat_2</span>,
                              in_state_first <span class="op">=</span> <span class="st">"gg_Nucb2_EDTA_0.001"</span>,
                              chosen_state_first <span class="op">=</span> <span class="st">"gg_Nucb2_EDTA_25"</span>,
                              out_state_first <span class="op">=</span> <span class="st">"gg_Nucb2_EDTA_1440"</span>,
                              in_state_second <span class="op">=</span> <span class="st">"gg_Nucb2_CaCl2_0.001"</span>,
                              chosen_state_second <span class="op">=</span> <span class="st">"gg_Nucb2_CaCl2_25"</span>,
                              out_state_second <span class="op">=</span> <span class="st">"gg_Nucb2_CaCl2_1440"</span><span class="op">)</span>                             

<span class="co"># theoretical comparison plot - relative values</span>
<span class="fu"><a href="../reference/comparison_plot.html">comparison_plot</a></span><span class="op">(</span>calc_dat <span class="op">=</span> <span class="va">calc_dat_2</span>,
                theoretical <span class="op">=</span> <span class="cn">TRUE</span>,
                relative <span class="op">=</span> <span class="cn">TRUE</span>,
                state_first <span class="op">=</span> <span class="st">"Nucb2 Factor 1"</span>,
                state_second <span class="op">=</span> <span class="st">"Nucb2 Factor 2"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Theoretical fraction exchanged in \nstate comparison in 25 min time"</span><span class="op">)</span></code></pre></div>
<p><img src="HaDeX_files/figure-html/unnamed-chunk-36-1.png" width="672"></p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># experimental comparison plot - relative values </span>
<span class="fu"><a href="../reference/comparison_plot.html">comparison_plot</a></span><span class="op">(</span>calc_dat <span class="op">=</span> <span class="va">calc_dat_2</span>,
                theoretical <span class="op">=</span> <span class="cn">FALSE</span>,
                relative <span class="op">=</span> <span class="cn">TRUE</span>,
                state_first <span class="op">=</span> <span class="st">"Nucb2 Factor 1"</span>,
                state_second <span class="op">=</span> <span class="st">"Nucb2 Factor 2"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Fraction exchanged in \nstate comparison in 25 min time"</span><span class="op">)</span></code></pre></div>
<p><img src="HaDeX_files/figure-html/unnamed-chunk-36-2.png" width="672"></p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># theoretical comparison plot - absolute values</span>
<span class="fu"><a href="../reference/comparison_plot.html">comparison_plot</a></span><span class="op">(</span>calc_dat <span class="op">=</span> <span class="va">calc_dat_2</span>,
                theoretical <span class="op">=</span> <span class="cn">TRUE</span>,
                relative <span class="op">=</span> <span class="cn">FALSE</span>,
                state_first <span class="op">=</span> <span class="st">"Nucb2 Factor 1"</span>,
                state_second <span class="op">=</span> <span class="st">"Nucb2 Factor 2"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Theoretical fraction exchanged in \nstate comparison in 25 min time"</span><span class="op">)</span></code></pre></div>
<p><img src="HaDeX_files/figure-html/unnamed-chunk-36-3.png" width="672"></p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># experimental comparison plot - absolute values </span>
<span class="fu"><a href="../reference/comparison_plot.html">comparison_plot</a></span><span class="op">(</span>calc_dat <span class="op">=</span> <span class="va">calc_dat_2</span>,
                theoretical <span class="op">=</span> <span class="cn">FALSE</span>,
                relative <span class="op">=</span> <span class="cn">FALSE</span>,
                state_first <span class="op">=</span> <span class="st">"Nucb2 Factor 1"</span>,
                state_second <span class="op">=</span> <span class="st">"Nucb2 Factor 2"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Fraction exchanged in \nstate comparison in 25 min time"</span><span class="op">)</span></code></pre></div>
<p><img src="HaDeX_files/figure-html/unnamed-chunk-36-4.png" width="672"></p>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># theoretical Woods plot - relative values</span>
<span class="fu"><a href="../reference/woods_plot.html">woods_plot</a></span><span class="op">(</span>calc_dat <span class="op">=</span> <span class="va">calc_dat_2</span>,
           theoretical <span class="op">=</span> <span class="cn">TRUE</span>,
           relative <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Theoretical fraction exchanged between states in 25 min time"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_cartesian.html">coord_cartesian</a></span><span class="op">(</span>ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">.5</span>, <span class="fl">.7</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="HaDeX_files/figure-html/unnamed-chunk-36-5.png" width="672"></p>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># experimental Woods plot - relative values</span>
<span class="fu"><a href="../reference/woods_plot.html">woods_plot</a></span><span class="op">(</span>calc_dat <span class="op">=</span> <span class="va">calc_dat_2</span>,
           theoretical <span class="op">=</span> <span class="cn">FALSE</span>,
           relative <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Fraction exchanged between states in 25 min time"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_cartesian.html">coord_cartesian</a></span><span class="op">(</span>ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">.5</span>, <span class="fl">.7</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="HaDeX_files/figure-html/unnamed-chunk-36-6.png" width="672"></p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># theoretical Woods plot - absolute values</span>
<span class="fu"><a href="../reference/woods_plot.html">woods_plot</a></span><span class="op">(</span>calc_dat <span class="op">=</span> <span class="va">calc_dat_2</span>,
           theoretical <span class="op">=</span> <span class="cn">TRUE</span>,
           relative <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Theoretical fraction exchanged between states in 25 min time"</span><span class="op">)</span> </code></pre></div>
<p><img src="HaDeX_files/figure-html/unnamed-chunk-36-7.png" width="672"></p>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># experimental Woods plot - absolute values</span>
<span class="fu"><a href="../reference/woods_plot.html">woods_plot</a></span><span class="op">(</span>calc_dat <span class="op">=</span> <span class="va">calc_dat_2</span>,
           theoretical <span class="op">=</span> <span class="cn">FALSE</span>,
           relative <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Fraction exchanged between states in 25 min time"</span><span class="op">)</span> </code></pre></div>
<p><img src="HaDeX_files/figure-html/unnamed-chunk-36-8.png" width="672"></p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># quality control</span>
<span class="op">(</span><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/quality_control.html">quality_control</a></span><span class="op">(</span>dat <span class="op">=</span> <span class="va">dat_2</span>,
                           state_first <span class="op">=</span> <span class="st">"gg_Nucb2_EDTA"</span>,
                           state_second <span class="op">=</span> <span class="st">"gg_Nucb2_CaCl2"</span>, 
                           chosen_time <span class="op">=</span> <span class="fl">25</span>, 
                           in_time <span class="op">=</span> <span class="fl">0.001</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>##   out_time avg_err_state_first sd_err_state_first avg_err_state_second
## 1       60         0.005530984        0.002672025          0.007094412
## 2     1440         0.005308684        0.002471285          0.006814869
##   sd_err_state_second    avg_diff     sd_diff
## 1         0.002096056 0.009074865 0.002499717
## 2         0.002019402 0.008631836 0.002732523</code></pre>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># example quality control visualisation - relative values</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">result</span><span class="op">[</span><span class="va">result</span><span class="op">[</span><span class="st">"out_time"</span><span class="op">]</span><span class="op">&gt;=</span><span class="fl">1</span>,<span class="op">]</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html">geom_line</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">out_time</span>, y <span class="op">=</span> <span class="va">avg_err_state_first</span>, color <span class="op">=</span> <span class="st">"Average error (first state)"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html">geom_line</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">out_time</span>, y <span class="op">=</span> <span class="va">avg_err_state_second</span>, color <span class="op">=</span> <span class="st">"Average error (second state)"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_x_log10</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"log(time) [min]"</span>, y <span class="op">=</span> <span class="st">"Average uncertainty"</span>, title <span class="op">=</span> <span class="st">"Uncertainty change"</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"bottom"</span>,
        legend.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="HaDeX_files/figure-html/unnamed-chunk-36-9.png" width="672"></p>
</div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-BouyssieHDXViewerinteractive3D2019">
<p>Bouyssié, David, Jean Lesne, Marie Locard-Paulet, Renaud Albigot, Odile Burlet-Schiltz, and Julien Marcoux. 2019. “HDX-Viewer: Interactive 3D Visualization of Hydrogen-Deuterium Exchange Data.” <em>Bioinformatics (Oxford, England)</em>, July. <a href="https://doi.org/10.1093/bioinformatics/btz550">https://doi.org/10.1093/bioinformatics/btz550</a>.</p>
</div>
<div id="ref-HoudeUtilityHydrogenDeuterium2011">
<p>Houde, Damian, Steven A. Berkowitz, and John R. Engen. 2011. “The Utility of Hydrogen/Deuterium Exchange Mass Spectrometry in Biopharmaceutical Comparability Studies.” <em>Journal of Pharmaceutical Sciences</em> 100 (6): 2071–86. <a href="https://doi.org/10.1002/jps.22432">https://doi.org/10.1002/jps.22432</a>.</p>
</div>
<div id="ref-HourdelMEMHDXinteractivetool2016">
<p>Hourdel, Véronique, Stevenn Volant, Darragh P. O’Brien, Alexandre Chenal, Julia Chamot-Rooke, Marie-Agnès Dillies, and Sébastien Brier. 2016. “MEMHDX: An Interactive Tool to Expedite the Statistical Validation and Visualization of Large HDX-MS Datasets.” <em>Bioinformatics</em> 32 (22): 3413–9. <a href="https://doi.org/10.1093/bioinformatics/btw420">https://doi.org/10.1093/bioinformatics/btw420</a>.</p>
</div>
<div id="ref-jcgm:2008:EMDG">
<p>Joint Committee for Guides in Metrology. 2008. “JCGM 100: Evaluation of Measurement Data - Guide to the Expression of Uncertainty in Measurement.” JCGM.</p>
</div>
<div id="ref-LauDeuterossoftwarerapid2019">
<p>Lau, Andy M. C., Zainab Ahdash, Chloe Martens, and Argyris Politis. 2019. “Deuteros: Software for Rapid Analysis and Visualization of Data from Differential Hydrogen Deuterium Exchange-Mass Spectrometry.” <em>Bioinformatics (Oxford, England)</em>, January. <a href="https://doi.org/10.1093/bioinformatics/btz022">https://doi.org/10.1093/bioinformatics/btz022</a>.</p>
</div>
<div id="ref-MassonRecommendationsperforminginterpreting2019">
<p>Masson, Glenn R., John E. Burke, Natalie G. Ahn, Ganesh S. Anand, Christoph Borchers, Sébastien Brier, George M. Bou-Assaf, et al. 2019. “Recommendations for Performing, Interpreting and Reporting Hydrogen Deuterium Exchange Mass Spectrometry (HDX-MS) Experiments.” <em>Nature Methods</em> 16 (7): 595–602. <a href="https://doi.org/10.1038/s41592-019-0459-y">https://doi.org/10.1038/s41592-019-0459-y</a>.</p>
</div>
<div id="ref-PascalHDXWorkbenchSoftware2012">
<p>Pascal, Bruce D., Scooter Willis, Janelle L. Lauer, Rachelle R. Landgraf, Graham M. West, David Marciano, Scott Novick, Devrishi Goswami, Michael J. Chalmers, and Patrick R. Griffin. 2012. “HDX Workbench: Software for the Analysis of H/D Exchange MS Data.” <em>Journal of the American Society for Mass Spectrometry</em> 23 (9): 1512–21. <a href="https://doi.org/10.1007/s13361-012-0419-6">https://doi.org/10.1007/s13361-012-0419-6</a>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Weronika Puchala, Michal Burdukiewicz.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
